src/integrationTest/java/uk/gov/hmcts/darts/testutils/IntegrationBase.java

```diff
@@ -13,28 +19,92 @@
 import org.springframework.test.context.DynamicPropertyRegistry;
 import org.springframework.test.context.DynamicPropertySource;
 import org.testcontainers.containers.GenericContainer;
+import uk.gov.hmcts.darts.common.repository.AutomatedTaskRepository;
+import uk.gov.hmcts.darts.common.repository.EventHandlerRepository;
+import uk.gov.hmcts.darts.common.repository.RetentionPolicyTypeRepository;
+import uk.gov.hmcts.darts.common.repository.SecurityGroupRepository;
+import uk.gov.hmcts.darts.common.repository.UserAccountRepository;
 import uk.gov.hmcts.darts.test.common.LogUtil;
 import uk.gov.hmcts.darts.test.common.MemoryLogAppender;
 import uk.gov.hmcts.darts.testutils.stubs.DartsDatabaseStub;
 
 import java.util.List;
 
 /**
- * Base class for integration tests running with H2 in Postgres compatibility mode.
- * This class also starts a containerised Redis instance
+ * Base class for integration tests running with H2 in Postgres compatibility mode.<br>
+ * This class also starts a containerised Redis instance.<br>
+ *<br>
+ * To optimise tests total execution time, the below setup has been introduced:
+ * <ul>
+ *  <li>
+ *     predefined test data created by Liquibase (user accounts, security groups, event handlers, etc...) is not guaranteed
+ *     to be deleted and recreated in between test classes execution
+ *  </li>
+ *  <li>
+ *     test data created by tests is deleted before each test. Tables with predefined test data have only rows with id >= SEQUENCE_START_VALUE deleted,
+ *     i.e. the data created by tests, not Liquibase
+ *  </li>
+ *  <li>
+ *     sequences are reset to either SEQUENCE_START_VALUE or their initial value (depending on the type of data, if predefined or not) before each test
+ *  </li>
+ * </ul>
+ *<br>
+ * Based on the above, please follow the following recommendations when writing integration tests:
+ * <ul>
+ *  <li>
+ *     do not permanently modify predefined test data (e.g. changing a security group global flag, disabling an automated task, etc...)
+ *  </li>
+ *  <li>
+ *     setup test data in a @BeforeEach rather than a @BeforeAll
+ *  </li>
+ *  <li>
+ *      when creating test data with manually assigned ids use ids >= SEQUENCE_START_VALUE so that data is automatically deleted
+ *  </li>
+ * </ul>
  */
-@AutoConfigureWireMock(files = "file:src/integrationTest/resources/wiremock")
+@AutoConfigureWireMock(port = 0, files = "file:src/integrationTest/resources/wiremock")
 @SpringBootTest
+@Slf4j
 @ActiveProfiles({"intTest", "h2db", "in-memory-caching"})
 public class IntegrationBase {
 
+    private static final int SEQUENCE_START_VALUE = 500;
+
+    private static final List<String> SEQUENCES_NO_RESET = List.of(
+        "revinfo_seq"
+    );
+
+    private static final List<String> SEQUENCES_RESET_FROM = List.of(
+        "usr_seq",
+        "grp_seq",
+        "aut_seq",
+        "rpt_seq",
+        "evh_seq"
+    );
```

Comment:
With this, I suppose we'll have two ways of resetting the database state, depending on the nature of the table:
1. For tables with standing data, we delete all where `id >= sequence`
2. For tables without standing data, we delete all (existing functionality in `DartsDatabaseStub`)

Just a thought - but is there a reason for restricting which sequences we reset under (1)? I'm wondering if that mechanism could also be used to replace (2), then we don't need to worry about the distinction between standing data and non-standing data. This would probably need to happen in a specific order to avoid FK constrain violations.

I think this may help with maintainability of the class, as it would no longer have to track which tables contain "standing" data per Flyway.

Reply:
As discussed with Chris, wha we could do to unify the deletion approach is to have ALL the table sequences start from SEQUENCE_START_VALUE. This way all the tables will be deleted with mechanism 1).
I would have to spend some more time fixing some upcoming failing tests though.